src/controllers/loginController.ts

```diff
@@ -65,71 +65,66 @@ async function saveToken(username, token) {
   }
 }
 
-function censorEmail(email) {
-  const username = email.split('@')[0];
-  const domainSegments = email.split('@')[1].split('.');
-  const topLevelDomain = domainSegments[domainSegments.length - 1];
-  const emailHost = domainSegments.slice(0, -1).join('');
-
-  function censorWord(str) {
-    return str.length <= 5
-      ? str[0] + '*'.repeat(str.length - 1)
-      : str[0] + '*'.repeat(str.length - 2) + str.slice(-1);
-  }
-
-  return `${censorWord(username)}@${censorWord(emailHost)}.${censorWord(
-    topLevelDomain
-  )}`;
-}
-
 export async function getLogin(req, res) {
   renderLogin(req, res, {});
 }
 
 export async function postLogin(req, res) {
   const nextParam = req.query.next ? `?next=${req.query.next}` : '';
-  const { username, useSecondaryEmail } = req.body;
+  const { emailInput } = req.body;
+  const emailSplit = emailInput.split('@');
 
-  if (username === undefined || !/^[a-z0-9_-]+\.[a-z0-9_-]+$/.test(username)) {
+  if (!emailInput) {
     req.flash(
       'error',
-      "L'email renseigné n'a pas le bon format. Il doit contenir des caractères alphanumériques en minuscule et un '.'.<br /><i>Exemple : charlotte.duret</i>"
+      `Aucune adresse renseginée.`
     );
     return res.redirect(`/login${nextParam}`);
   }
 
+  let username;
+  let secondaryEmail;
+
+  if (emailSplit[1] === config.domain) {
```

Comment:
Potentiellement le résultat de `emailSplit` ne contiendra pas deux éléments, ce qui lancerait un out of bounds dans `[1]`.